src/finch/autoschedule/optimize.py

```diff
@@ -181,6 +184,108 @@ def propagate_fields(root: LogicNode) -> LogicNode:
     return _propagate_fields(root, fields={})
 
 
+def _tuple_diff(x1: tuple, x2: tuple) -> tuple:
+    return tuple([x for x in x1 if x not in x2])
+
+
+def _intersect(x1: tuple, x2: tuple) -> tuple:
+    return tuple(x for x in x1 if x in x2)
```

Comment:
Should these two also go into your `_util.py`?

Reply:
Good point - I moved them there.